gio/src/output_stream.rs

```diff
@@ -668,6 +668,30 @@ impl<T: IsA<OutputStream>> io::Write for OutputStreamWrite<T> {
         to_std_io_result(result)
     }
 
+    fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
+        let result = self
+            .0
+            .as_ref()
+            .write_all(buf, crate::Cancellable::NONE)
+            .and_then(|(_, e)| e.map(Err).unwrap_or(Ok(())));
+        to_std_io_result(result)
+    }
+
+    #[cfg(any(feature = "v2_60", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_60")))]
+    fn write_vectored(&mut self, bufs: &[io::IoSlice<'_>]) -> io::Result<usize> {
+        let vectors = bufs
+            .iter()
+            .map(|v| OutputVector::new(&**v))
+            .collect::<smallvec::SmallVec<[_; 2]>>();
```

Comment:
OOC, why do you use 2 here and in the other places?

Reply:
Random choice for a simple ring buffer usage, should I increase the number? Not sure what a good value would be.

Reply:
I don't know. So far I always used this API from C with exactly 2 too :) Let's keep it at that then.
I was just asking because making it e.g. 4 or 16 is also not really going to cause problems, these things are small.